src/coreComponents/physicsSolvers/surfaceGeneration/EmbeddedSurfaceGenerator.cpp

```diff
@@ -109,123 +109,354 @@ void EmbeddedSurfaceGenerator::initializePostSubGroups()
+    embSurfNodeManager.compressRelationMaps();
   } );
+}
+
+void EmbeddedSurfaceGenerator::initializePostInitialConditionsPreSubGroups()
+{}
+
+void EmbeddedSurfaceGenerator::propagationStep( DomainPartition & domain,
+                                                R1Tensor & currentTip,
+                                                R1Tensor & targetTip,
+                                                localIndex & tipElementIndex )
+{
+  GEOSX_MARK_FUNCTION;
+  // std::cout<<"propagation step"<<std::endl;
+  // std::cout<<"currentTip_x: "<<currentTip[0]<<std::endl;
+  // std::cout<<"currentTip_y: "<<currentTip[1]<<std::endl;
+  // std::cout<<"targetTip_x: "<<targetTip[0]<<std::endl;
+  // std::cout<<"targetTip_y: "<<targetTip[1]<<std::endl;
+  localIndex er = 0; //should be the element region number
+  localIndex esr = 0; //should be the element subregion number
+  // Get geometric object manager
+  GeometricObjectManager & geometricObjManager = GeometricObjectManager::getInstance();
+  // create BoundedPlane from currentTip to newTip
+  std::unique_ptr< BoundedPlane > newFracturePlane = std::make_unique< BoundedPlane >( currentTip[0], currentTip[1], targetTip[0],
+                                                                                       targetTip[1], "newFracturePlane", &geometricObjManager );
+  // BoundedPlane * newFracturePlane = new BoundedPlane( currentTip[0], currentTip[1], targetTip[0],
+  //                                                     targetTip[1], "newFracturePlane", &geometricObjManager );
+  // Get meshLevel
+  MeshLevel & meshLevel = domain.getMeshBody( 0 ).getBaseDiscretization();
+  // Get managers
+  ElementRegionManager & elemManager = meshLevel.getElemManager();
+  NodeManager & nodeManager = meshLevel.getNodeManager();
+  EmbeddedSurfaceNodeManager & embSurfNodeManager = meshLevel.getEmbSurfNodeManager();
+  EdgeManager & edgeManager = meshLevel.getEdgeManager();
+  FaceManager & faceManager = meshLevel.getFaceManager();
+  arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD > const & nodesCoord = nodeManager.referencePosition();
+  // Get EmbeddedSurfaceSubRegions
+  SurfaceElementRegion & embeddedSurfaceRegion = elemManager.getRegion< SurfaceElementRegion >( this->m_fractureRegionName );
+  EmbeddedSurfaceSubRegion & embeddedSurfaceSubRegion = embeddedSurfaceRegion.getSubRegion< EmbeddedSurfaceSubRegion >( 0 );
+  localIndex localNumberOfSurfaceElems = embeddedSurfaceSubRegion.size(); //this should not be zero
+  NewObjectLists newObjects; //this should probably not be re-initialized
+  // begin geometric operations
+  real64 const planeCenter[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( newFracturePlane->getCenter() );
+  real64 const normalVector[3] = LVARRAY_TENSOROPS_INIT_LOCAL_3( newFracturePlane->getNormal() );
+  // Initialize variables
+  globalIndex nodeIndex;
+  integer isPositive, isNegative;
+  real64 distVec[ 3 ];
+
+  //COPYING FROM EXISTING FUNCTION - NEEDS CLEANING
```

Comment:
can we do some cleaning. Also, I think we need to add an explanation of the algorithm. it gets hard to follow. I would also break this down in a few smaller functions to help understanding what's happening.